### DISCUSSION TITLE
Issue number should be displayed alongside the title

### DISCUSSION THREAD

**Discussion by polevaultweb** [community/community#5904] [Projects and Issues] [answered]
When viewing and discussing a project board with multiple team members, people have started to refer to issues by the number in the left hand column:

![image](https://user-images.githubusercontent.com/1770201/135442662-166e30db-2e7f-4b7c-ae91-200b285237a4.png)

However, this isn't static for an issue and changes when items are moved around.

Historically, when discussing an issue we would refer to it from the issue number as shorthand, instead of reading out the whole title. But the issue/pr number is not visible in the board table view currently.

Would be great to see the issue number either before or after the issue title - eg.

![image](https://user-images.githubusercontent.com/1770201/135443060-40337839-1809-4d4b-8b8b-16f53ac5fb1d.png)

**Comment by KindBill** [comment]
One thing to consider though is that a project can consist of issues within multiple repositories. This may lead to issues with the same issue #. 

I do like the idea of the issue # in front of the title since the title of the issue can be cut off. Or perhaps it should exist within its own column?

**Reply by arfanliaqat** [reply]
@KindBill i think that would be real bummer where same numbered issue are shown from different repos, may be we can allow a column for repo name as well to elevate this? +1 for showing issue number

**Reply by RESLibber** [reply]
+1 for showing the issue number in a separate sortable column

**Reply by glasser** [reply]
There already is a column available for repo name. But not for issue number.

**Comment by alexcnichols** [comment]
👋🏼 Improvements for displaying the issue number in the project table, and more, are coming soon! Check out this roadmap item for more and subscribe to be notified if/when things change. https://github.com/github/roadmap/issues/283 :octocat:

**Reply by attieretief** [reply]
@alexcniAlex this feature is not on the referenced roadmap item anymore and can't find it anywhere in the public roadmap either...real bummer as it was scheduled for Dec 2021...final thing holding its back migrating from a different project tool

**Reply by alexcnichols** [reply]
Yeah, looks like this didn't quite make it into the improvements in December. Adding @azenMatt in case he has anything he'd like to share about if/when this is coming.

**Reply by azenMatt** [reply]
Thanks Alex - this is still on our backlog as an improvement we want to make but I don't have a timeline to share just yet

**Reply by attieretief** [reply]
It's really sad not to see this anywhere on the roadmap anymore. Why is this not a priority?
@azenMatt

**Reply by chriskuech** [reply]
Ya kind of confused how this has 10x the number of votes as some more difficult issues yet isn't addressed.  Seems like a super easy win to prioritize.

**Reply by attieretief** [reply]
> Thanks Alex - this is still on our backlog as an improvement we want to make but I don't have a timeline to share just yet

@azenMatt is there really still no plans for addressing this simple, very useful, highly requested, obvious improvement? 🥴

**Reply by LukasSchauffele** [reply]
@azenMatt any updates on this one?

**Reply by SpencerKaiser** [reply]
@alexcnichols bump on the above - any updates since October? The issue number is pretty critical and it's surprising that wasn't included with the initial release 😕

**Reply by azenMatt** [reply]
👋 yes! this just shipped 😄  

![image](https://user-images.githubusercontent.com/7584089/187922815-d42b11be-8f8f-4dcf-aa93-da82e8fb3fce.png)

**Comment by polevaultweb** [comment]
Thanks @alexcnichols 💪

**Comment by mikegillan** [comment]
@alexcnichols hope you don't mind but I'm adding my vote to this feature request. I see the roadmap you linked is closed, but no mention of this feature. Hope it didn't slip through the cracks. Thanks! 😸

**Reply by polevaultweb** [reply]
Good catch @mikegillan!

**Comment by elopatin-uc3** [comment]
@azenMatt @alexcnichols Adding another vote for surfacing actual issue numbers, regardless of the repo they stem from. We've adopted use of the project table and a variety of views. But not having those issue numbers (nor being able to sort by them) has been a bummer. That said, thank you all of your work and for letting us know this functionality is still on the backlog. Look forward to seeing it in the future!

**Reply by attieretief** [reply]
@alexcnichols @azenMatt why does this seem to have died a death...a sad one?

**Comment by mikegillan** [comment]
I can’t believe this (seemingly) simple and much-requested feature has
fallen by the wayside. Any update would be most appreciated as the
work-round is clumsy and results in commit reference errors.

**Comment by JJTech0130** [comment]
Here's a userscript I just made to do this: https://gist.github.com/JJTech0130/09c0e1122f5e06dd341fd3e348085d00
![Screen Shot 2022-07-20 at 10 12 46 AM](https://user-images.githubusercontent.com/53275876/180004141-c3184208-164c-4d21-9bf8-4d4efab646cd.png)

**Reply by JJTech0130** [reply]
If people are actually interested in having this as a userscript, I can add more features (like making it a sortable column). If not, this at least proves it’s super easy to implement.

**Reply by JJTech0130** [reply]
(Right now it’s purely cosmetic, it doesn’t affect the actual issue title or the sorting)

**Reply by JJTech0130** [reply]
I made another version which changes the number on the side:
![Screen Shot 2022-07-20 at 1 31 31 PM](https://user-images.githubusercontent.com/53275876/180046039-4a0037b4-373f-4e7f-b69f-303b9a2f5d40.png)
Which version is better?

**Reply by RESLibber** [reply]
Does this option remove the count of tickets on the left of the first option?

**Reply by JJTech0130** [reply]
The second version (which I just added to the gist) replaces the arbitrary leftmost numbers with the issue numbers. 
EDIT: I still haven't found a way to make them sortable, because the actual sorting code underneath still thinks in terms of the old names, the userscript is just cosmetics rn.

**Reply by RESLibber** [reply]
I like the second version best

**Comment by mikegillan** [comment]
Same, second version looks better. Those arbitrary GH numbers are useless
and would cause confusion.

**Comment by azenMatt** [comment, marked as answer]
Hi everyone, thanks for all your feedback and patience on this one, 

Items numbers has just shipped! 

![image](https://user-images.githubusercontent.com/7584089/187923016-ac6d5ae5-b233-433b-ac4e-2de55249f8cc.png)

**Comment by RESLibber** [comment]
@azenMatt I appreciate that you are listening and making updates.  Is it possible to get the numbers in their own field so we can sort on the number?

**Reply by azenMatt** [reply]
👀 cc @rileybroughten who is the PM for this

**Reply by kfogel** [reply]
Thank you so much for shipping this!

FWIW, I second @RESLibber's suggestion (though just having the numbers at all is already a huge win, even not in their own distinct field).

**Reply by rileybroughten** [reply]
Thanks @RESLibber! Will take this feedback back to the team 👍

**Reply by euanmillar** [reply]
@rileybroughten do you know if @RESLibber 's suggestion has been actioned?  It would be great to choose if issue number is always after the title or in it's own column.

**Reply by fjkish** [reply]
Putting the number in the title is ok but it would be better as other have mentioned as being a separate column.
Also these numbers when part of the title don't seem to be included in if you download to TSV, which then means you've lost the main linkage back to the issue.

**Comment by SjorsO** [comment]
Glad to see that it now puts the issue number behind the title. 

One point of feedback, could the issue number also be a link to the issue, just like the title is?

For example, I'd like to hide the title and just show the number. I can do that by making the column small. But now I can't navigate to the issue, because only the title is a link:

<img width="138" alt="image" src="https://user-images.githubusercontent.com/7202674/188357805-0423ff66-c828-42ac-973b-32f415ed133e.png">
